lib/active_model_serializers/adapter/json_api/pagination_links.rb

```diff
@@ -35,19 +35,21 @@ def as_json
         private
 
         def pages_from
-          return {} if collection.total_pages <= FIRST_PAGE
-
           {}.tap do |pages|
-            pages[:self] = collection.current_page
+            pages[:self]  = collection.current_page
+            pages[:first] = FIRST_PAGE
+            pages[:last]  = collection.total_pages
 
-            unless collection.current_page == FIRST_PAGE
-              pages[:first] = FIRST_PAGE
-              pages[:prev]  = collection.current_page - FIRST_PAGE
-            end
+            if collection.total_pages > 0
+              unless collection.current_page == FIRST_PAGE
+                pages[:prev] = collection.current_page - FIRST_PAGE
+              end
 
-            unless collection.current_page == collection.total_pages
-              pages[:next] = collection.current_page + FIRST_PAGE
-              pages[:last] = collection.total_pages
+              unless collection.current_page == collection.total_pages
+                pages[:next] = collection.current_page + FIRST_PAGE
+              end
+            else
+              pages[:last] = FIRST_PAGE
             end
           end
```

Comment:
@bf4 in your code snippet, why do you think `context` responds to `url_for`? 

```
context.url_for(action: :index, params: params)
```
I did notice that `url_helpers` seem to be available as an instance variable, however:

```
[7] pry(#<ActiveModelSerializers::Adapter::JsonApi::PaginationLinks>)> context.url_helpers
NoMethodError: undefined method `url_helpers' for #<ActiveModelSerializers::SerializationContext:0x007ff3af0e2338>
from (pry):7:in `url_for_page'
[8] pry(#<ActiveModelSerializers::Adapter::JsonApi::PaginationLinks>)> context.instance_variable_get :@url_helpers
=> #<Module:0x007ff3aedee488>
[9] pry(#<ActiveModelSerializers::Adapter::JsonApi::PaginationLinks>)> _.lm
=> [:url_options, :url_for, :_routes, :optimize_routes_generation?, :included, :append_features, :class_methods]
```